test/chainrules.jl

```diff
@@ -2,6 +2,7 @@
     @testset "BlockDiagonal" begin
         x = [randn(1, 2), randn(2, 2)]
         test_rrule(BlockDiagonal, x)
+        test_rrule(BlockDiagonal, x; output_tangent=Tangent{BlockDiagonal}(;blocks=x))
```

Comment:
Does `rand_tangent` not effectively cover this - as it will call `ProjectTo` returning a `BlockDiagonal`. In which case, is the test case that needs to be added now actually a dense matrix because the behaviour there has changed?

Reply:
Oh ignore me, you are testing the constructor.

Reply:
Before this PR, `test_rrule(BlockDiagonal, x)` was the only test and that tested a `Tangent` output tangent via `rand_tangent`. Using CRTU 1.0 the `rand_tangent` returns a `BlockDiagonal` instead (after defining `ProjectTo`). So the extra test tests the previous behaviour, while the existing test tests something new.
We could also test with a `Matrix` tangent, but maybe in a future PR?

Reply:
Right, I understand. I said the previous thinling that `rand_tangent` (then in FiniteDifferences) would be creating a `Dense` matrix.